corehq/apps/case_importer_v1/tasks.py

```diff
@@ -82,11 +83,18 @@ def _submit_caseblocks(domain, case_type, caseblocks):
                 )
             else:
                 properties = set().union(*map(lambda c: set(c.dynamic_case_properties().keys()), cases))
-                add_inferred_export_properties.delay(
-                    'CaseImporter',
-                    domain,
-                    case_type,
-                    properties,
+                if case_type and len(properties):
+                    add_inferred_export_properties.delay(
+                        'CaseImporter',
+                        domain,
+                        case_type,
+                        properties,
+                    )
+                _soft_assert = soft_assert(notify_admins=True)
+                _soft_assert(
+                    len(properties) == 0 and not case_type,
```

Comment:
definitely not blocking but I was thinking this might be more readable if it was
```python
if case_type and len(properties):
  add_inferred_export_properties.delay(...)
else:
  _soft_assert(len(properties) == 0, ...)
```
feel free to update if you agree or ignore. i think what you have is functionally equivalent.